### ISSUE TITLE
Report on Lion Optimizer

### ISSUE THREAD

**Issue by Mut1nyJD** [lucidrains/denoising-diffusion-pytorch#170]
First run looks very promising although I implemented it myself but it basically near identical to what you did. I just divided the learn rate by 3 though. 

But so far it seems to converge nicely and probably faster too than with AdamW. With AdamW I actually had an issue recently when I changed the linear attention to full attention on the two coarsest layers in down and up sample branch of the UNet. There I had to restart training twice because it got stuck with NaNs

**Comment by lucidrains** [comment]
@Mut1nyJD nice! so in the paper, in the limitations section, they said it didn't really help improve anything for Imagen, something noteworthy

in my mind, if this technique is not worse than adam, it is worth using as it is simpler, slightly more memory efficient. i think this technique should also require a learning rate cooldown too.

**Comment by lucidrains** [comment]
> But so far it seems to converge nicely and probably faster too than with AdamW. With AdamW I actually had an issue recently when I changed the linear attention to full attention on the two coarsest layers in down and up sample branch of the UNet. There I had to restart training twice because it got stuck with NaNs

are you saying that you are seeing anecdotally more stability during training?

**Comment by Mut1nyJD** [comment]
@lucidrains 
> are you saying that you are seeing anecdotally more stability during training?

Yes but I need a slightly longer run and I've got only one sample point so not enough for a definitive conclusion

**Comment by lucidrains** [comment]
@Mut1nyJD thanks for sharing!

**Comment by Zhaoyi-Yan** [comment]
Emm, for me, lion works very bad when I directly replace adamw with lion, the performance degrades significantly.

**Comment by lucidrains** [comment]
@Zhaoyi-Yan how high is your batch size and what was your learning rate?

**Comment by Mut1nyJD** [comment]
It is the standard lr divided by 3 which is probably 1:3e-05 and batchsize is 56 I definitely see a faster convergence will post samples when I am back from holiday

**Comment by lucidrains** [comment]
@Mut1nyJD some people are reporting that although it has faster convergence, it does not generalize as well. would be curious about what your final FID scores are

**Comment by Mut1nyJD** [comment]
> @Mut1nyJD some people are reporting that although it has faster convergence, it does not generalize as well. would be curious about what your final FID scores are

So I am back first numbers also show better FID scores than AdamW in my generated samples but I have to do a more excessive test on later checkpoints with AdamW I guess as Lion converges quicker. But after 100k iterations Lion optimizer wins with a 20 point margin.
